src/shared/platform/posix/nacl_host_desc.c

```diff
@@ -183,7 +186,14 @@ uintptr_t NaClHostDescMap(struct NaClHostDesc *d,
    */
   tmp_prot = host_prot & ~PROT_EXEC;
   need_exec = (0 != (PROT_EXEC & host_prot));
-  map_addr = mmap(start_addr, len, tmp_prot, host_flags, desc, offset);
+  if(!start_addr){
+    NaClLog(LOG_FATAL,
+            "NaClHostDescMap: start_addr cannot be NULL.\n");
+  }
+  whichcage = d ? d->cageid : 0;
```

Comment:
I think a bit of commenting here would be helpful for the future, I understand what's going on but we had talked about it.

Reply:
Okay will do.

Reply:
Added!